### PR TITLE
[Indexer] Stream full blocks instead of transactions

### PR DESCRIPTION
**Pull request by bowenyang007**
### Description
Stream full blocks in the format of
```
DMLOG BLOCK_START <height> <version>
DMLOG TRX <transaction proto base64>
DMLOG TRX <transaction proto base64>
…
DMLOG BLOCK_END
```
Added validation for the transactions to ensure correctness before we stream the data out. If anything is wrong with any of the transactions within a block, we fail the entire block and do not stream anything. 

### Test Plan
cargo test
ensure `cargo run -p aptos -- node run-local-testnet --with-faucet` works as expected

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2748)

### CODE REVIEW

File: sf-stream/src/metrics.rs [hunk 1]
```diff
@@ -4,10 +4,10 @@
 use aptos_metrics_core::{register_int_counter, IntCounter};
 use once_cell::sync::Lazy;
 
-pub static TRANSACTIONS_SENT: Lazy<IntCounter> = Lazy::new(|| {
+pub static BLOCKS_SENT: Lazy<IntCounter> = Lazy::new(|| {
     register_int_counter!(
-        "aptos_sf_stream_transactions_sent_count",
-        "Transactions converted and printed out to stdout, picked up by the StreamingFast indexer",
+        "aptos_sf_stream_blocks_sent_count",
+        "Blocks converted and printed out to stdout, picked up by the StreamingFast indexer",
```
Comment (CapCap) [inline sf-stream/src/metrics.rs:10]: Let’s keep transactions sent, and also add a new counter for blocks pls :-)

File: sf-stream/src/runtime.rs [hunk 2]
```diff
@@ -109,80 +93,83 @@ impl SfStreamer {
...
-                            "[sf-stream] oldest txn version is {} but requested version is {}",
-                            oldest_version, &self.current_version
-                        );
-                    }
-                }
-                None => {
-                    return vec![];
-                }
-            },
+    pub async fn convert_next_block(&mut self) -> Vec<TransactionPB> {
```
Comment (CapCap) [inline sf-stream/src/runtime.rs:110]: This is going to do a Lot more non-grouped reads than the previous implementation

File: sf-stream/src/runtime.rs [hunk 3]
```diff
@@ -191,43 +178,108 @@ impl SfStreamer {
...
+
+    pub async fn print_block(&self, encoded_txns: &Vec<String>, block_start_version: u64) {
+        println!(
+            "DMLOG BLOCK_START {} {}",
+            self.current_block_height, block_start_version
+        );
+        for encoded_txn in encoded_txns {
+            println!("DMLOG TRX {}", encoded_txn);
+        }
+        println!("DMLOG BLOCK_END");
```
Comment (CapCap) [inline sf-stream/src/runtime.rs:282]: Pls include block number here too, for sanity?

File: sf-stream/src/runtime.rs (hunk 2 above)
Comment (bowenyang007) [reply sf-stream/src/runtime.rs:110]: What do you mean by this? We are still batch querying?

**Comment by bowenyang007** [comment]
deprecate in favor of https://github.com/aptos-labs/aptos-core/pull/2821

**Pull request closed without merging**